R/utils.R

```diff
@@ -140,3 +140,51 @@ resolve_modules_datanames <- function(modules, datanames, join_keys) {
     modules
   }
 }
+
+check_modules_datanames <- function(modules, datanames) {
+  recursive_check_datanames <- function(modules, datanames) {
+    # check teal_modules against datanames
+    if (inherits(modules, "teal_modules")) {
+      sapply(modules$children, function(module) recursive_check_datanames(module, datanames = datanames))
+    } else {
+      extra_datanames <- setdiff(modules$datanames, c("all", datanames))
+      if (length(extra_datanames)) {
+        sprintf(
+          "- Module '%s' uses different datanames than available in the 'data': (%s) not in (%s)",
+          modules$label,
+          toString(dQuote(extra_datanames, q = FALSE)),
+          toString(dQuote(datanames, q = FALSE))
+        )
+      }
+    }
+  }
+  check_datanames <- unlist(recursive_check_datanames(modules, datanames))
+  if (length(check_datanames)) {
+    paste(check_datanames, collapse = "\n")
+  } else {
+    TRUE
+  }
+}
+
+
+check_filter_datanames <- function(filters, datanames) {
```

Comment:
```suggestion
#' Check `datanames` in filters
#'
#' This function check `datanames` in filters correspond to those in `data`,
#' returning character vector with error messages or TRUE if all checks pass.
#'
#' @param filters (`teal_slices`) object
#' @param datanames (`character`) names of datasets available in the `data` object
#'
#' @return A character vector containing error messages or TRUE if validation passes.
#' @keywords internal
check_filter_datanames <- function(filters, datanames) {
```

Adding documentation text